tests/tests_unit/test_utils/test_json.py

```diff
@@ -0,0 +1,52 @@
+import json
+from decimal import Decimal
+
+import pytest
+
+from cognite.client.data_classes._base import CogniteObject
+from cognite.client.utils import _json
+from cognite.client.utils._importing import local_import
+
+
+class TestJsonDumpDefault:
+    def test_json_serializable_Decimal(self):
+        with pytest.raises(TypeError):
+            json.dumps(Decimal(1))
+
+        assert _json.dumps(Decimal(1))
+
+    def test_json_not_serializable_sets(self):
+        with pytest.raises(TypeError):
+            json.dumps({1, 2})
+        with pytest.raises(TypeError):
+            json.dumps({1, 2})
+
+    @pytest.mark.dsl
+    def test_json_serializable_numpy(self):
+        np = local_import("numpy")
+        arr = np.array([1.2, 3.4], dtype=np.float32)
+        with pytest.raises(TypeError):
+            json.dumps(arr)
+        with pytest.raises(TypeError):
+            json.dumps(arr[0])
+        with pytest.raises(TypeError):  # core sdk makes it hard to serialize np.ndarray
```

Comment:
Not sure who wrote this comment originally ...but this can be trivially fixed with another instance check in `_default_json_encoder` (not that I see a use case for it)